File: LeetCode/TwoPointers/java/isSubsequence.java
package LeetCode.TwoPointers.java;

record isSubsequenceRecord(String s, String t, boolean expected) {
}

public class isSubsequence {
    public boolean isSubsequenceSolution(String s, String t) {
        int sp = 0;
        int tp = 0;

        while (sp < s.length() && tp < t.length()) {
            if (s.charAt(sp) == t.charAt(tp)) {
                sp += 1;
            }
            tp += 1;
        }

        return sp == s.length() ? true : false;
    }

    public static void main(String[] args) {
        isSubsequenceRecord input = new isSubsequenceRecord("abc", "ahbgdc", true);
        isSubsequence.testSolution(input);

        input = new isSubsequenceRecord("axc", "ahbgdc", false);
        isSubsequence.testSolution(input);
    }

    public static void testSolution(isSubsequenceRecord input) {
        System.out.println("Input: s: " + input.s());
        System.out.println("Input: t: " + input.t());
        System.out.println("Expected: " + input.expected());
        boolean val = new isSubsequence().isSubsequenceSolution(input.s(), input.t());
        System.out.println("Result: " + val + ", " + (val == input.expected() ? "Correct" : "Wrong"));
        System.out.println("-".repeat(50));
    }
}

File: LeetCode/TwoPointers/python/isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        sp = 0
        tp = 0

        while sp < len(s) and tp < len(t):
            if s[sp] == t[tp]:
                sp += 1
            
            tp += 1

        return True if sp == len(s) else False

    @staticmethod
    def testSolution(input) -> None:
        print("Input: s: {}".format(input["s"]))
        print("Input: t: {}".format(input["t"]))
        print("Expected: {}".format(input["expected"]))
        val = Solution().isSubsequence(input["s"], input["t"])
        print("Result: {}, {}".format(val, ("Correct" if val == input["expected"] else "Wrong")))
        print("-" * 50)



if __name__ == "__main__":
    input = {"s":"abc", "t":"ahbgdc", "expected": True}
    Solution.testSolution(input)

    input = {"s":"axc", "t":"ahbgdc", "expected": False}
    Solution.testSolution(input)